### PR TITLE
Updated v1 to support latest version of go-tuf

### PR DESCRIPTION
**Pull request by kommendorkapten**
#### Summary
Per the latest [TUF updates](https://github.com/sigstore/root-signing/pull/1164) in Sigstore Public Good instance, the key type is changing for the TUF keys, to keep cosign v1 continue to work I've updated to the latest go-tuf version.

#### Release Note
* Updated go-tuf version to v0.7.0
* Updated sigstore/sigstore to v1.8.0

#### Documentation
N/A

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -100,7 +100,7 @@ lint: golangci-lint ## Run golangci-lint linter
 	$(GOLANGCI_LINT_BIN) run -n
 
 test:
-	go test $(shell go list ./... | grep -v third_party/)
+	GODEBUG=x509sha1=1 go test $(shell go list ./... | grep -v third_party/)
```
Comment (kommendorkapten) [inline Makefile:103]: This is needed as some certs are using SHA-1.

**Comment by jku** [comment]
I'm not super familiar with cosign (like what v1 is) but does this mean upgrading cosign to use the go-tuf rewrite?

**Comment by kommendorkapten** [comment]
> EDIT: I suppose 0.7 is the release before the rewrite?

Correct! This is the last version that is API compatible.

**Comment by haydentherapper** [comment]
LGTM, just need to rebase off release-1.13 and merge into that branch instead, which should hopefully resolve test failures.

**Comment by kommendorkapten** [comment]
Rebased on `release-1.13` now.

**Pull request closed without merging**